src/plugins/vis_type_timeseries/server/lib/vis_data/request_processors/table/date_histogram.ts

```diff
@@ -58,19 +65,38 @@ export const dateHistogram: TableRequestProcessorsFunction = ({
   };
 
   const overwriteDateHistogramForEntireTimerangeMode = () => {
-    const intervalString = `${to.valueOf() - from.valueOf()}ms`;
+    const metricAggs = Object.values<string>(METRIC_AGGREGATIONS);
+    let intervalForEntireTimerangeMode;
 
     panel.series.forEach((column) => {
+      isAggSupported(column.metrics);
+
       const aggRoot = calculateAggRoot(doc, column);
 
-      overwrite(doc, `${aggRoot}.timeseries.auto_date_histogram`, {
-        field: timeField,
-        buckets: 1,
-      });
+      if (column.metrics.every((metric) => metricAggs.includes(metric.type))) {
+        overwrite(doc, `${aggRoot}.timeseries.auto_date_histogram`, {
+          field: timeField,
+          buckets: 1,
+        });
+
+        intervalForEntireTimerangeMode = `${to.valueOf() - from.valueOf()}ms`;
+      } else {
+        overwrite(doc, `${aggRoot}.timeseries.date_histogram`, {
+          field: timeField,
+          min_doc_count: 0,
+          time_zone: timezone,
+          extended_bounds: {
+            min: from.valueOf(),
+            max: to.valueOf(),
+          },
+          ...dateHistogramInterval(intervalString),
+        });
+        intervalForEntireTimerangeMode = intervalString;
+      }
 
       overwrite(doc, aggRoot.replace(/\.aggs$/, '.meta'), {
         ...meta,
-        intervalString,
+        intervalForEntireTimerangeMode,
```

Comment:
why it was renamed only for table? I prefer to save `intervalString` ? especially if that param used for both modes

Reply:
It's my mistake. We should use intervalString. Fixed.